File: (7 kyu) Shortest Word/(7 kyu) Shortest Word.js
// #1
// function findShort(s) {
//   const arr = s.split(' ');
//   let smallest = arr[0];
//   for (let i = 0; i < arr.length; i++) {
//     if (arr[i].length < smallest.length) {
//       smallest = arr[i];
//     }
//   }
//   return smallest.length;
// }
// #2
// const findShort = (s) =>
//   s
//     .split(' ')
//     .sort((a, b) => b.length - a.length)
//     .pop().length;
// #3
const findShort = (s) => Math.min(...s.split` `.map((w) => w.length));

File: (7 kyu) Shortest Word/(7 kyu) Shortest Word.java
// 1 Plain solution
import java.util.Arrays;

public class Kata {
  public static int findShort(String s) {
    String[] arr = s.split(" ");
    int min = s.length();
    
    for (String w : arr) {
      if (w.length() < min) {
        min = w.length();
      }
    }
    
    return min;
  }
}

// 2 Optimized solution
import java.util.*;

public class Kata {
  public static int findShort(String s) {
    int min = s.length();
    
    for (String w : s.split(" ")) {
      min = Math.min(w.length(), min);
    }
    
    return min;
  }
}

// 3 Clever solution
import java.util.*;

public class Kata {
  public static int findShort(String s) {
    return Stream.of(s.split(" "))
      .mapToInt(String::length)
      .min()
      .getAsInt();
  }
}
